api/app/models/env_canada.py

```diff
@@ -181,8 +181,9 @@ def mark_prediction_model_run_processed(session: Session,
         year=now.year,
         month=now.month,
         day=now.day,
-        hour=hour, tzinfo=datetime.timezone.utc)
-    prediction_run_timestamp = adjust_model_day(prediction_run_timestamp, hour)
+        hour=now.hour, tzinfo=datetime.timezone.utc)
+    prediction_run_timestamp = adjust_model_day(prediction_run_timestamp, model_run_hour)
```

Comment:
So the bug was `adjust_model_day` never did the job here because `prediction_run_timestamp`'s hour was set with `model_run_hour` leading to comparing the same thing